addons/mrp_account/models/mrp_workorder.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, _
+from odoo.tools import float_is_zero
+
+
+class MrpWorkorder(models.Model):
+    _inherit = 'mrp.workorder'
+
+    mo_analytic_account_line_id = fields.Many2one('account.analytic.line')
+    wc_analytic_account_line_id = fields.Many2one('account.analytic.line')
+
+    def _compute_duration(self):
+        super()._compute_duration()
+        for order in self:
+            order._create_or_update_analytic_entry()
```

Comment:
Do we need to create in advance analytic line? It would be more efficient to compute them once at the end of the workorder

Reply:
Or maybe we should add a flag for prevision and real analytic

Reply:
I will check with mgm